DataFormats/HGCalReco/interface/Trackster.h

```diff
@@ -28,6 +28,30 @@ namespace ticl {
 
     edm::ProductID seedID;
     int seedIndex;
+
+    // regressed energy
+    float regressed_energy;
+
+    // types considered by the particle identification
+    enum class ParticleType {
+      photon = 0,
+      electron,
+      muon,
+      neutral_pion,
+      charged_hadron,
+      neutral_hadron,
+      ambiguous,
+      unknown,
+    };
+
+    // trackster ID probabilities
+    std::array<float, 8> id_probabilities;
+
+    // convenience method to return the ID probability for a certain particle type
+    inline float id_probability(ParticleType type) const {
+      // probabilities are stored in the same order as defined in the ParticleType enum
+      return id_probabilities[(int)type];
```

Comment:
non-trivial methods are not really expected for plain structs in data formats.
Perhaps it's time to make this a normal class with private/encapsulated data members

Reply:
I agree. But making all members private and adding getters/setters would result in a lot of additional changes. Maybe we should create another PR for this?

Reply:
Ok to postpone to another PR
Issue #28049 has been created to keep track of it